qiita_db/commands.py

```diff
@@ -348,77 +352,63 @@ def update_preprocessed_data_from_cmd(sl_out_dir, study_id, ppd_id=None):
             "%s" % (sl_out_dir, ', '.join(missing_files)))
 
     # Get the preprocessed data to be updated
-    study = Study(study_id)
-    ppds = study.preprocessed_data()
-    if not ppds:
-        raise ValueError("Study %s does not have any preprocessed data")
-
-    if ppd_id:
-        if ppd_id not in ppds:
-            raise ValueError("The preprocessed data %d does not exist in "
-                             "study %d. Available preprocessed data: %s"
-                             % (ppd_id, study_id, ', '.join(map(str, ppds))))
-        ppd = PreprocessedData(ppd_id)
-    else:
-        ppd = PreprocessedData(sorted(ppds)[0])
-
-    # We need to loop through the fps list to get the db filepaths that we
-    # need to modify
-    fps = defaultdict(list)
-    for fp_id, fp, fp_type in sorted(ppd.get_filepaths()):
-        fps[fp_type].append((fp_id, fp))
-
-    fps_to_add = []
-    fps_to_modify = []
-    keys = ['preprocessed_fasta', 'preprocessed_fastq', 'preprocessed_demux',
-            'log']
-
-    for key in keys:
-        if key in fps:
-            db_id, db_fp = fps[key][0]
-            fp_checksum = compute_checksum(new_fps[key])
-            fps_to_modify.append((db_id, db_fp, new_fps[key], fp_checksum))
+    with TRN:
+        study = Study(study_id)
+        ppds = study.preprocessed_data()
+        if not ppds:
+            raise ValueError("Study %s does not have any preprocessed data")
+
+        if ppd_id:
+            if ppd_id not in ppds:
+                raise ValueError(
+                    "The preprocessed data %d does not exist in "
+                    "study %d. Available preprocessed data: %s"
+                    % (ppd_id, study_id, ', '.join(map(str, ppds))))
+            ppd = PreprocessedData(ppd_id)
         else:
```

Comment:
Oops, I did not see this one.

This is GitHub messing up. That else (in my version) is actually the one in line 362 in this line. If you check line 363 here you will see that is line 364 in my version...